sample_building/sample_building.py

```diff
@@ -564,3 +566,97 @@ def azimuth_to_antenna_offset(beamdir, main_antenna_count, interferometer_antenn
         beams_antenna_phases.append(phase_array)
 
     return beams_antenna_phases
+
+
+def write_samples_to_file(txrate, txctrfreq, list_of_pulse_dicts,
+                          file_path, main_antenna_count, final_rx_sample_rate):
+    """
+    Write the samples and transmitted metadata to a json file for use in testing.
+
+    :param txrate: The rate at which these samples will be transmitted at.
+    :param txctrfreq: The centre frequency that the N200 is tuned to (and will mix with
+     these samples).
+    :param list_of_pulse_dicts: The list of all pulse dictionaries for pulses included
+    in this sequence. Pulse dictionaries have all metadata and the samples for the
+    pulse.
+    :param file_path: location to place the json file.
+    :param main_antenna_count: The number of antennas available for transmitting on.
+    :param final_rx_sample_rate: The final sample rate after decimating on the receive
+    side.
```

Comment:
A three sentence or so description of the method you use to loop over sequences, and the structure of the output data file would make this code very skimmable in the future. Overall, pretty easy to understand.